components/datetime/src/calendar.rs

```diff
@@ -730,56 +652,212 @@ where
     Ok(payload)
 }
 
+#[cfg(any(feature = "datagen", feature = "experimental"))]
+trait Sealed {}
+
+/// A collection of marker types associated with all calendars.
+#[allow(private_bounds)] // sealed
+#[cfg(any(feature = "datagen", feature = "experimental"))]
+pub trait CalMarkers<M>: Sealed
```

Comment:
thought: this trait feels backwards to me but i also think this is the best way to do this.

suggestion: more docs maybe, covering how it gets used?